From a519f1bda444a0c6c01464296a052d9a05e1d9c4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Emilio=20Cobos=20=C3=81lvarez?= <emilio@crisal.io>
Date: Sat, 4 Apr 2020 23:45:18 +0200
Subject: [PATCH] Actually use the temp dir for temporary compiled objects.

If the cbindgen_output is an absolute path, then join() will just return it.

Really fixes #499.
---
 tests/tests.rs | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tests/tests.rs b/tests/tests.rs
index 503b9f52..3e8bc51d 100644
--- a/tests/tests.rs
+++ b/tests/tests.rs
@@ -74,7 +74,10 @@ fn compile(
         Language::C => env::var("CC").unwrap_or_else(|_| "gcc".to_owned()),
     };
 
-    let mut object = tmp_dir.join(cbindgen_output);
+    let file_name = cbindgen_output
+        .file_name()
+        .expect("cbindgen output should be a file");
+    let mut object = tmp_dir.join(file_name);
     object.set_extension("o");
 
     let mut command = Command::new(cc);
